From 3ec14992acb06caa2b3a26633844603589db7f8a Mon Sep 17 00:00:00 2001
From: Jan Kratochvil <jan.kratochvil@redhat.com>
Date: Fri, 18 May 2012 17:44:31 +0000
Subject: [PATCH] gdb/
 	* auto-load.c (show_auto_load_safe_path): Accept any combination of
 	DIRNAME_SEPARATOR and IS_DIR_SEPARATOR for wild-match.

---
 gdb/ChangeLog   |    5 +++++
 gdb/auto-load.c |   10 +++++++++-
 2 files changed, 14 insertions(+), 1 deletions(-)

Index: gdb-7.2/gdb/auto-load.c
===================================================================
--- gdb-7.2.orig/gdb/auto-load.c
+++ gdb-7.2/gdb/auto-load.c
@@ -241,7 +241,15 @@ static void
 show_auto_load_safe_path (struct ui_file *file, int from_tty,
 			  struct cmd_list_element *c, const char *value)
 {
-  if (strcmp (value, "/") == 0)
+  const char *cs;
+
+  /* Check if user has entered either "/" or for example ":".
+     But while more complicate content like ":/foo" would still also
+     permit any location do not hide those.  */
+
+  for (cs = value; *cs && (*cs == DIRNAME_SEPARATOR || IS_DIR_SEPARATOR (*cs));
+       cs++);
+  if (*cs == 0)
     fprintf_filtered (file, _("Auto-load files are safe to load from any "
 			      "directory.\n"));
   else
